javascript/javascript2/week3/warmUp.js

```diff
@@ -0,0 +1,192 @@
+
+const funnyJokes = ['Q: What do you call sad coffee? A: Despresso.', "Q: What's the best Beatles song? A: Latte Be!", "Q: Why do they call coffee mud? A: Because it was ground a couple of minutes ago.", "Q: How are coffee beans like kids? A: They're always getting grounded!", "Q: What's it called when you steal someone's coffee? A: Mugging!", "Q: How does a tech guy drink coffee? A: He installs Java!", "Q: How did the hipster burn his tongue? A: He drank his coffee before it was cool.", "Q: Why are Italians so good at making coffee? A: Because they know how to espresso themselves.", "Q: How are coffee beans like kids? A: They're always getting grounded!", "If the local coffee shop has awarded you 'Employee of the Month' and you don't even work there, you may be drinking too much coffee."]
+const badJokes = ["I saw this advert in a window that said: “Television for sale, £1, volume stuck on full.” I thought, “I can’t turn that down.”", "What's Forrest Gump's password? 1forrest1", "What's Harry Potter's favourite method of getting down a hill? Walking... JK, Rolling.", "What did the clock do when it was hungry? It went back four seconds.", "What has two legs and bleeds? Half a cat.", "What did Batman say to Robin before they got in the car? Robin, get in the car.", "Why do scuba divers fall backwards off of the boat? Because if they fell forward, they'd still be on the boat.", "Did you hear about the two thieves who stole a calendar? They each got six months.", 'And so the Lord said unto John, "Come forth; and receive eternal life." But John came fifth and won a toaster.', 'I still remember the last thing my grandfather said before kicking the bucket: "Hey, wanna see how far I can kick this bucket?"']
+
+const random = Math.floor(Math.random() * 10)
+
+function logFunnyJoke(){
+    console.log(funnyJokes[random])
+}
+
+function logBadJoke(){
+    console.log(badJokes[random])
+}
+
+function jokeCreator(shouldTellFunnyJoke) {
+    if(shouldTellFunnyJoke){
+        logFunnyJoke()
+    }
+    else{
+        logBadJoke()
+    }
+}
+
+jokeCreator(true)
+jokeCreator(false)
+
+// 9. Additional (I want to do for try)
+
+const jokesButton = document.getElementById('jokesButton')
+const jokeParagraph = document.getElementById('jokes')
+
+function jokeCreator2() {
+    const shouldTellFunnyJoke2 = confirm('Do you want funny joke ?')
+    const random2 = Math.floor(Math.random() * 10)
+    if (shouldTellFunnyJoke2) {
+        jokeParagraph.textContent = funnyJokes[random2]
+    }
+    else {
+        jokeParagraph.textContent = badJokes[random2]
+    }
+}
+
+jokesButton.addEventListener('click', jokeCreator2)
+
+
+// Function as a variable
+
+// Create an array with 3 items. All items should be functions. Iterate through the array and call all the functions.
+
+const functionArray = [function(){console.log('Function One')},function(){console.log('Function Two')},function (){console.log('Function Three')}]
+
+functionArray.forEach(item => item())
```

Comment:
To make this a bit more readable, I'd define the function above and then add them to the array:
```js
function one() {
  console.log('Function One')
}
function two() {
  console.log('Function Two')
}
function three() {
  console.log('Function Three')
}
const functionArray = [one, two, three]

functionArray.forEach(item => item())
```